commercetools-models/src/main/java/io/sphere/sdk/customers/commands/CustomerSignInCommand.java

```diff
@@ -38,19 +38,26 @@
     private final String password;
     @Nullable
     private final String anonymousCartId;
+    @Nullable
+    private final String anonymousId;
 
-    private CustomerSignInCommand(final String email, final String password, @Nullable final String anonymousCartId) {
+    private CustomerSignInCommand(final String email, final String password, @Nullable final String anonymousCartId, @Nullable final String anonymousId) {
         this.email = email;
         this.password = password;
         this.anonymousCartId = anonymousCartId;
+        this.anonymousId = anonymousId;
     }
 
     public static CustomerSignInCommand of(final String email, final String password) {
         return of(email, password, null);
     }
 
     public static CustomerSignInCommand of(final String email, final String password, @Nullable final String anonymousCartId) {
-        return new CustomerSignInCommand(email, password, anonymousCartId);
+        return new CustomerSignInCommand(email, password, anonymousCartId, null);
+    }
+
+    public CustomerSignInCommand withAnonymousId(@Nullable final String anonymousId) {
+        return new CustomerSignInCommand(email, password, anonymousCartId, anonymousId);
```

Comment:
historical grown technical dept, I did not assume that the number of possible parameters will grow and ofEmailPasswordAnonymousId sounded weird. With withers we can easily grow.

I don't like `public static CustomerSignInCommand of(final String email, final String password, @Nullable final String anonymousCartId)` anymore. I would just use the "of" with two parameters and also use for anonymousCartId a wither.